### PR TITLE
Add process dynamics class for external contacts

### PR DESCRIPTION
**Pull request by isorrentino**
This PR introduced a process model for the external contacts defined as 

$$
\dot{f} = -k f 
$$

In addition, the PR changes the logic to define a name for the dynamics. Indeed, the name of the dynamics has to match the name `ukf_name` defined in the `model` configuration file. This solves the problem described in https://github.com/ami-iit/bipedal-locomotion-framework/issues/760 which was causing a segmentation fault when using the ft frame as the external contact frame. 

cc @GiulioRomualdi @S-Dafarra

### CODE REVIEW

File: src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModel.h [hunk 1]
```diff
@@ -33,15 +33,16 @@ namespace RobotDynamicsEstimator
...
+    std::string ukfName; /**< Name of the dynamics used for this sensor. */
+    int frameIndex;      /**< Index of the sensor in the URDF. */
 };
 
 /**
- * FT describes the force/torque sensors of the submodels.
+ * FTSENSOR describes the force/torque sensors of the submodels.
  */
-struct FT : Sensor
+struct FTSENSOR : Sensor
```
Comment (GiulioRomualdi) [inline src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModel.h:45]: This should be `FTSensor`

File: src/Estimators/src/SubModel.cpp [hunk 2]
```diff
@@ -347,26 +355,21 @@ bool RDE::SubModelCreator::createSubModels(
...
-        return false;
-    }
+        if (!group->getParameter("ukf_names", ukfNames))
+        {
+            blf::log()->error("{} The parameter handler could not find 'ukf_names' in the "
+                              "configuration file for the group {}.",
+                              logPrefix,
+                              groupName);
+            return false;
+        }
```
Comment (GiulioRomualdi) [inline src/Estimators/src/SubModel.cpp:366]: You can remove `blf::`

File: src/Estimators/tests/RobotDynamicsEstimator/RobotDynamicsEstimationTest.cpp [hunk 3]
```diff
@@ -293,9 +426,13 @@ TEST_CASE("RobotDynamicsEstimator Test")
...
         REQUIRE(((output.tau_m - output.tau_F) - dataset.expectedTauj.row(sample).transpose()).isZero(0.1));
-        for (auto const& [key, value] : dataset.fts)
+        for (int idx = 0; idx < sensors["ft"].size(); idx++)
+        {
+            REQUIRE((output.ftWrenches[sensors["ft"][idx].ukfSensorName] - dataset.fts[sensors["ft"][idx].sensorName].row(sample).transpose()).isZero(0.1));
+        }
+        for (int idx = 0; idx < sensors["contact"].size(); idx++)
         {
-            REQUIRE((output.ftWrenches[key] - value.row(sample).transpose()).isZero(0.1));
+            REQUIRE((output.contactWrenches[sensors["contact"][idx].ukfSensorName]).isZero(5.0));
```
Comment (GiulioRomualdi) [inline src/Estimators/tests/RobotDynamicsEstimator/RobotDynamicsEstimationTest.cpp:435]: 5.0 as tolerance is a lot https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#af36014ec300f53a65083057ed4e89822

File: src/Estimators/tests/RobotDynamicsEstimator/RobotDynamicsEstimationTest.cpp [hunk 4]
```diff
@@ -293,9 +426,13 @@ TEST_CASE("RobotDynamicsEstimator Test")
         REQUIRE((output.tau_F - dataset.expectedTauF.row(sample).transpose()).isZero(0.1));
         REQUIRE((output.tau_m - dataset.expectedTaum.row(sample).transpose()).isZero(0.1));
         REQUIRE(((output.tau_m - output.tau_F) - dataset.expectedTauj.row(sample).transpose()).isZero(0.1));
-        for (auto const& [key, value] : dataset.fts)
+        for (int idx = 0; idx < sensors["ft"].size(); idx++)
+        {
+            REQUIRE((output.ftWrenches[sensors["ft"][idx].ukfSensorName] - dataset.fts[sensors["ft"][idx].sensorName].row(sample).transpose()).isZero(0.1));
```
Comment (GiulioRomualdi) [inline src/Estimators/tests/RobotDynamicsEstimator/RobotDynamicsEstimationTest.cpp:431]: You can use `isApprox` (https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#ae8443357b808cd393be1b51974213f9c)

```suggestion
            REQUIRE((output.ftWrenches[sensors["ft"][idx].ukfSensorName].isApprox( dataset.fts[sensors["ft"][idx].sensorName].row(sample).transpose()));
```

**Comment by isorrentino** [comment]
I pushed the changes after your review. I saw that the CI is failing but the errors are not related to the code of this PR.

**Comment by isorrentino** [comment]
Hi @GiulioRomualdi, I just rebased this branch on master. Feel free to review it again (if needed) or merge it.

**Pull request merged**